kore/src/Logic.hs

```diff
@@ -34,8 +36,18 @@ mapLogicT ::
     (forall x. m x -> n x) ->
     LogicT m a ->
     LogicT n a
-mapLogicT nat acts = (lift . nat) (observeAllT acts) >>= scatter
+mapLogicT nat acts = hoistLogicT nat acts
 {-# INLINE mapLogicT #-}
+#if !MIN_VERSION_logict(0,8,0)
```

Comment:
From @jberthold : we should remove this once we are definitely on the `0.8` version